pkg/multicluster/access/kubeconfig_secret.go

```diff
@@ -0,0 +1,97 @@
+package access
+
+import (
+	kubev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+)
+
+// KubeConfigSecretLabel is applied to a kubernetes secret to indicate that the secret contains a kubeconfig.
+const KubeConfigSecretLabel = "solo.io/kubeconfig"
+
+// KubeConfig wraps kubernetes KubeConfigs and also indicates the name of the cluster the KubeConfig belongs to.
+// Used to persist KubeConfigs to kubernetes secrets.
+type KubeConfig struct {
+	// the actual KubeConfig
+	Config api.Config
+	// expected to be used as an identifier string for a cluster
+	// stored as the key for the KubeConfig data in a kubernetes secret
+	Cluster string
+}
+
+// KubeConfigToSecret converts a single KubeConfig to a secret with the provided name and namespace.
+func KubeConfigToSecret(name string, namespace string, kc *KubeConfig) (*kubev1.Secret, error) {
+	return KubeConfigsToSecret(name, namespace, []*KubeConfig{kc})
+}
+
+// KubeConfigsToSecret converts a list of KubeConfigs to a KubeConfig secret with the provided name and namespace.
+func KubeConfigsToSecret(name string, namespace string, kcs []*KubeConfig) (*kubev1.Secret, error) {
+	secretData := map[string][]byte{}
+	for _, kc := range kcs {
+		rawKubeConfig, err := clientcmd.Write(kc.Config)
+		if err != nil {
+			return nil, FailedToConvertKubeConfigToSecret(err)
+		}
+		if _, exists := secretData[kc.Cluster]; exists {
+			return nil, DuplicateClusterName(kc.Cluster)
+		}
+		secretData[kc.Cluster] = rawKubeConfig
+	}
+	return &kubev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels:    map[string]string{KubeConfigSecretLabel: "true"},
+			Name:      name,
+			Namespace: namespace,
+		},
+		Type: kubev1.SecretTypeOpaque,
```

Comment:
have we considered using `KubeConfigSecretLabel` as the `Type` instead of Opaque? I think this is the intended convention for the Type field